Ignore Navigator touch events for background buttons

corbt opened the pull request:
Currently, if the Navigator with the default `NavigatorNavigationBar` has two scenes on its `routeStack` and the bottom one defines a `RightButton` but the top one doesn't, a touch to the location of the underlying  `RightButton` will trigger its action.

This fix checks if the button's opacity is set to 0 (indicating it has been transitioned off the scene and shouldn't be interacted with) and ignores touch events if so.

corbt commented:
Sorry, this was off the wrong branch, updating.

manask88 commented:
This solution works for me. Hey @corbt , can you please rebase your branch?  .

corbt commented:
@manask88 rebased

torifat commented:
I'm facing the same problem :(

mkonicek commented:
Looks good to me.

@ericvicenti Should we merge this?

ide commented:
This is a good bug to fix but make sure to test it in a variety of scenarios in the UIExplorer, especially with resetting the route stack when push/pop aren't called.

ericvicenti commented:
This looks reasonable. Please rebase and do a final round of testing, then I'll merge it

corbt commented:
Ok, I've rebased. I also added some more logic to deal with the the routeStack being reset without passing through a transition. I've tested in the UIExplorer with transitions and resets and this seems to cover all cases now.

cc @ericvicenti

corbt commented on Libraries/CustomComponents/Navigator/NavigatorNavigationBar.js:
```diff
@@ -145,6 +145,7 @@ var NavigatorNavigationBar = React.createClass({
       var component = this._components[componentName].get(this.props.navState.routeStack[index]);
       var props = this._getReusableProps(componentName, index);
       if (component && interpolate[componentName](props.style, amount)) {
+        props.pointerEvents = props.style.opacity === 0 ? 'none' : 'box-none';
```
I'm not really happy with this line because the check happens on every iteration in a tight loop. But I'm not sure where a better place to put it would be; the NavigationBar's `props` don't appear to be updated ever with the final route after a transition has taken place, so I can't make it contingent on that.

ide replied:
This is a cheap check. The main thing we want to avoid is actually setting the native property every frame if we can help it, which in this case we should, since most of the time we're not changing the value of the pointerEvents property.

facebook-github-bot-9 commented:
@ericvicenti would you mind taking a look at this pull request? It's been a while since the last commit was reviewed.

ide commented:
I tested this in the UIExplorer's Navigator demo and it appears to work fine with normal push/pop transitions and reset-transitions as well. Stress-testing it, it behaves the same as master (with bugs -- not this commits fault AFAICT).

ide commented:
@facebook-github-bot shipit

facebook-github-bot-3 commented:
@ericvicenti would you mind taking a look at this pull request? It's been a while since the last commit was reviewed.

corbt commented:
(I rebased on master just in case that was causing a problem with the import)

@facebook-github-bot shipit
